### ISSUE TITLE
Replace semantic-release-firefox with semantic-release-firefox-add-on in community plugin list

### ISSUE THREAD

**Issue by iamogbz** [tophat/semantic-release-firefox-add-on#13]
Replace [`felixfbecker/semantic-release-firefox`](https://github.com/felixfbecker/semantic-release-firefox) on the [semantic-release plugin list](https://github.com/semantic-release/semantic-release/blob/master/docs/extending/plugins-list.md).

`felixfbecker/semantic-release-firefox` is currently [broken](https://github.com/felixfbecker/semantic-release-firefox/issues). It relies on running puppeteer to interact with the web interface of the mozilla webstore to sign and upload extensions, this makes it brittle as the website is subject to change.

`tophat/semantic-release-firefox-add-on` uses [Mozilla's web extension api](https://github.com/mozilla/web-ext)

**Comment by iamogbz** [comment]
I think I'll just submit a pull request to [semantic-release](https://github.com/semantic-release/semantic-release) to replace `semantic-release-firefox` with `semantic-release-firefox-add-on`. Reason being our plugin is in active development and tested and supports the most recent version of Mozilla's web extension api.

Further more it solves both https://github.com/felixfbecker/semantic-release-firefox/issues/38 and https://github.com/felixfbecker/semantic-release-firefox/issues/36 which have been open since march last year.

**Comment by dbasilio** [comment]
Done! It got merged into master last night!

**Comment by dbasilio** [comment]
PR for recording sake https://github.com/semantic-release/semantic-release/pull/1416

**Comment by felixfbecker** [comment]
> It relies on running puppeteer to interact with the web interface of the mozilla webstore to sign and upload extensions, this makes it brittle as the website is subject to change.

Afaik the AMO API does not support uploading sources, which according to the policies are required if any bundler is used to bundle the source code. Therefor Puppeteer seems the only possible option. Has this changed?
FWIW the AMO submit website has barely seen any changes in the past years (only changes were to the sign in flow).

**Comment by iamogbz** [comment]
> Afaik the AMO API does not support uploading sources, which according to the policies are required if any bundler is used to bundle the source code. Therefor Puppeteer seems the only possible option. Has this changed?'

I think including source files in the [`sourceDir`](https://github.com/mozilla/web-ext/blob/master/src/cmd/sign.js) option would be sufficient, see #178.

> FWIW the AMO submit website has barely seen any changes in the past years (only changes were to the sign in flow).

Even so, there's no guarantee it would remain the same vs the more stable api.

**Comment by felixfbecker** [comment]
> I think including source files in the sourceDir option would be sufficient, see #178.

In the [official documentation of `web-ext`](https://extensionworkshop.com/documentation/develop/web-ext-command-reference/#:~:text=Specifies%20the%20directory,If) `sourceDir` is only documented as an option for specifying sources when _building_ or _running_ an extension (using `web-ext build` and `web-ext run`). There is no mention of it being submitted to the store.

You can also look at the source code of `web-ext sign`: `sourceDir` is not passed to the `signAddon()` call in any way, only the path of the built XPI is. This is the API used by [`tophat/semantic-release-firefox-add-on`](https://github.com/tophat/semantic-release-firefox-add-on/blob/d67bcb0c1212119c2cd71ff9e1735774d16c58df/src/publish.js#L43).

https://github.com/mozilla/web-ext/blob/c866678c95bfb755fee190bb8dbe163d90aa6c9c/src/cmd/sign.js#L115-L127

Allowing to submit sources with `web-ext`, as opposed to just signing the extension, has been a long-time open issue at `web-ext` (https://github.com/mozilla/web-ext/issues/804), which is blocked by the AMO API not supporting this (https://github.com/mozilla/sign-addon/issues/409 https://github.com/mozilla/addons-server/issues/9913).

As far as I can tell, `tophat/semantic-release-firefox-add-on` therefor cannot publish extensions in a way that is compliant with Mozilla's policies if the package uses a bundling process, which is probably almost all Firefox extensions using semantic-release.

It's a fair argument that `felixfbecker/semantic-release-firefox` is prone to break if Mozilla updates the interface, but at least it's the only plugin to my knowledge that can actually publish extensions compliant with Mozilla's ToS. I therefor don't think it's justified to remove it from the community plugin list and replace it with the alternative.

**Comment by dbasilio** [comment]
At the time we tried to use `felixfbecker/semantic-release-firefox` it was broken and was not clear it would be fixed, so we built an alternative. `tophat/semantic-release-firefox-add-on` has been used by tophat for almost two years at this point successfully. Others have also adopted the plugin without issue and no one has raised issues on the repo with regards to it not working. 

We won't be offended if both plugins are listed and then users can choose which plugin they wish to use, however, you will want to raise that issue with semantic-release, not with us.
